Framework/DataHandling/src/SaveTBL.cpp

```diff
@@ -83,16 +83,28 @@ void SaveTBL::exec() {
         std::string groupHeading = columnHeadings[columnIndex];
         if (ws->getColumn(groupHeading)->type() != "int") {
           file.close();
-          remove(filename.c_str());
-          throw std::runtime_error(groupHeading +
-                                   " Column must be of type \"int\"");
+          int error = remove(filename.c_str());
+          if (error == 0)
```

Comment:
We throw in the event that remove was successful because the table was not formatted correctly, and we throw in the event remove was unsuccessful because we failed to remove the file. I hope this explains the code a bit more